Makefile.uk

```diff
@@ -0,0 +1,78 @@
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+################################################################################
+# App registration
+################################################################################
+$(eval $(call addlib,libembeddedgcov,$(CONFIG_LIBEMBEDDEDGCOV)))
+
+################################################################################
+# Sources
+################################################################################
+LIBEMBEDDEDGCOV_BRANCH = main
+LIBEMBEDDEDGCOV_URL = https://github.com/nasa-jpl/embedded-gcov.git
+LIBEMBEDDEDGCOV_PATCHDIR = $(LIBEMBEDDEDGCOV_BASE)/patches
+$(eval $(call clone,libembeddedgcov,$(LIBEMBEDDEDGCOV_URL),$(LIBEMBEDDEDGCOV_BRANCH)))
+$(eval $(call patch,libembeddedgcov,$(LIBEMBEDDEDGCOV_PATCHDIR),$(LIBEMBEDDEDGCOV_BASENAME)))
+
+################################################################################
+# Library sources
+################################################################################
+LIBEMBEDDEDGCOV_SRCS-y += $(LIBEMBEDDEDGCOV_ORIGIN)/code/gcov_public.c
+LIBEMBEDDEDGCOV_SRCS-y += $(LIBEMBEDDEDGCOV_ORIGIN)/code/gcov_gcc.c
+LIBEMBEDDEDGCOV_SRCS-y += $(LIBEMBEDDEDGCOV_ORIGIN)/code/gcov_printf.c
+
+################################################################################
+# Library includes
+################################################################################
+CINCLUDES-$(CONFIG_LIBEMBEDDEDGCOV) += -I$(LIBEMBEDDEDGCOV_ORIGIN)/code
+CXXINCLUDES-$(CONFIG_LIBEMBEDDEDGCOV) += -I$(LIBEMBEDDEDGCOV_ORIGIN)/code
+
+################################################################################
+# Library flags
+################################################################################
+
+CFLAGS-$(CONFIG_LIBEMBEDDEDGCOV) += -ftest-coverage -fprofile-arcs
+
+ifneq ($(CONFIG_LIBEMBEDDEDGCOV_OUTPUT_BINARY_FILE),)
+$(error Binary file output not implemented)
+endif
+ifneq ($(CONFIG_LIBEMBEDDEDGCOV_OUTPUT_BINARY_MEMORY),)
+$(error Memory output not implemented)
+endif
+
+LIBEMBEDDEDGCOV_CFLAGS-$(CONFIG_LIBEMBEDDEDGCOV_OUTPUT_SERIAL_HEXDUMP)     += -DGCOV_OPT_OUTPUT_SERIAL_HEXDUMP
+LIBEMBEDDEDGCOV_CFLAGS-$(CONFIG_LIBEMBEDDEDGCOV_OUTPUT_SERIAL_HEXDUMP)     += -Dgcov_printf=printf
```

Comment:
The gcov_printf function is not only used when the `CONFIG_LIBEMBEDDEDGCOV_OUTPUT_SERIAL_HEXDUMP` option is set, but also when the `GCOV_OPT_PRINT_STATUS` option is set. This can be seen in the original embedded-gcov code where the gcov_printf function is used when either of these options is defined. This means that even if the user selects a memory output, they may still want to see console information about the gcov run.

```
#if defined(GCOV_OPT_PRINT_STATUS) || defined(GCOV_OPT_OUTPUT_SERIAL_HEXDUMP)
    GCOV_PRINT_STR("Gcov End");
    GCOV_PRINT_STR("\n");
#endif
```

```suggestion
LIBEMBEDDEDGCOV_CFLAGS-y												   += -Dgcov_printf=printf
```

Reply:
Thanks, updated 👍🏼